### PR TITLE
Fix parsing of SwiftPackageManager dependencies including `macro` targets

### PR DESCRIPTION
**Pull request by jimmy-li-houzz**
Resolves https://github.com/tuist/tuist/issues/5307

### Short description 📝

Add a new enum case in `TargetType`, and make necessary changes to compile `TuistKit`.

### How to test the changes locally 🧐

Not sure. Didn't see any tests in `TuistSupportTests` relates to parsing Package.swift, and not sure how to add one

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Review by danyf90 — APPROVED** [review]
Can you also add a dependency with macors to the `projects/tuist/fixtures/app_with_spm_dependencies` fixture to make sure it generates and compiles correctly? 🙏

**Comment by jimmy-li-houzz** [comment]
Hey, I found another issue in this project `projects/tuist/fixtures/app_with_spm_dependencies`

1. `FirebaseApp.app()!` got a crash because we did not call `FirebaseApp.configure()`
2. We can't call `FirebaseApp.configure()` because we don't have a valid GoogleService-Info.plist

I think I will comment out all Firebase entry point for now.

**Comment by jimmy-li-houzz** [comment]
> Can you also add a dependency with macors to the `projects/tuist/fixtures/app_with_spm_dependencies` fixture to make sure it generates and compiles correctly? 🙏

Hi, I added a local macro package, but got and error when running tuist generate from [this guide](https://docs.tuist.io/contributors/get-started/).
```
Couldn't find target 'StringifyMacros_watchos' at '/Users/jimmyli/sources/tuist/projects/tuist/fixtures/app_with_spm_dependencies/StringifyMacro'
```

I have spent several hours but couldn't figure it out. Can you guide?

File: projects/tuist/fixtures/app_with_spm_dependencies/App/Sources/AppKit/AppKit.swift [hunk 1]
```diff
@@ -20,16 +20,16 @@ public enum AppKit {
...
         // Use FirebaseDatabase to make sure it links fine
-        Database.database(app: FirebaseApp.app()!).reference().setValue("value")
+//        Database.database(app: FirebaseApp.app()!).reference().setValue("value")
 
         // Use FirebaseCrashlytics to make sure it links fine
         _ = Crashlytics.crashlytics()
 
         // Use FirebaseFirestore to make sure it links fine
-        _ = Firestore.firestore()
+//        _ = Firestore.firestore()
```
Comment (danyf90) [inline projects/tuist/fixtures/app_with_spm_dependencies/App/Sources/AppKit/AppKit.swift:32]: I would keep them to make sure Firebase compiles fine.
If the app can't start because of some other misconfiguration we can try to fix it but not in this PR

File: projects/tuist/fixtures/app_with_spm_dependencies/Tuist/Dependencies.swift [hunk 2]
```diff
@@ -4,13 +4,14 @@ import ProjectDescriptionHelpers
 let packages: [Package] = [
     .package(url: "https://github.com/Alamofire/Alamofire", .upToNextMajor(from: "5.6.0")),
     .package(url: "https://github.com/facebook/facebook-ios-sdk", .upToNextMajor(from: "13.2.0")),
-    .package(url: "https://github.com/firebase/firebase-ios-sdk", .upToNextMajor(from: "9.3.0")),
+    .package(url: "https://github.com/firebase/firebase-ios-sdk", .upToNextMajor(from: "10.12.0")),
     .package(url: "https://github.com/pointfreeco/swift-composable-architecture", .upToNextMinor(from: "0.40.0")),
     .package(url: "https://github.com/iterable/swift-sdk", .upToNextMajor(from: "6.4.0")),
     .package(url: "https://github.com/Trendyol/ios-components", .revision("c9260bfe203a16a278eca5542c98455eece98aa4")),
-    .package(url: "https://github.com/stripe/stripe-ios", .upToNextMajor(from: "22.4.0")),
+    .package(url: "https://github.com/stripe/stripe-ios", .upToNextMajor(from: "23.11.1")),
```
Comment (danyf90) [inline projects/tuist/fixtures/app_with_spm_dependencies/Tuist/Dependencies.swift:11]: Why these changes? Let's try to keep changes related to this PR only

**Review by danyf90 — CHANGES_REQUESTED** [review]

File: projects/tuist/fixtures/app_with_spm_dependencies/Tuist/Dependencies.swift (hunk 2 above)
Comment (jimmy-li-houzz) [reply projects/tuist/fixtures/app_with_spm_dependencies/Tuist/Dependencies.swift:11]: Old version firebase won't compile with Xcode 15, and stripe has some shared dependency with firebase, has to update as well

**Review by danyf90 — APPROVED** [review]

**Review by pepicrft — APPROVED** [review]
